src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1038,15 +1038,21 @@ const Json::Value& CascadiaSettings::_GetDisabledProfileSourcesJsonObject(const
 }
 
 // Method Description:
-// - Create a backup file with the current contents, iff one does not exist
+// - Create a timestamped backup file with the current contents
 // Arguments:
 // - content - the content that we're writing to the backup file
 // - settingsPath - the path to the settings file that we're going to create a backup for
 // Return Value:
 // - <none>
 void CascadiaSettings::_WriteBackupFile(std::string_view content, const winrt::hstring settingsPath)
 {
-    const auto backupSettingsPath{ settingsPath + L".backup" };
+    // create a timestamped backup file
+    time_t timeStamp;
+    tm localTimeStamp;
+    time(&timeStamp);
+    localtime_s(&localTimeStamp, &timeStamp);
+    const auto backupSettingsPath{ fmt::format(L"{}{}{}{}{}{}.backup", settingsPath, localTimeStamp.tm_year, localTimeStamp.tm_mon, localTimeStamp.tm_mday, localTimeStamp.tm_hour, localTimeStamp.tm_min, localTimeStamp.tm_sec) };

```

Comment:
definitely don't do this! it'll turn jan 1 at 1:01:01 AM into "202111111" instead of "20210101010101"!

Reply:
you may wanna consider...


```
#include <fmt/chrono.h>
...
std::time_t timeStamp;
std::time(&timeStamp);
xxx = fmt::format(L"{}.{:%Y-%m-%dT%H:%M:%S}.backup", settingsPath, fmt::localtime(timeStamp));
```

(fmt has a strftime time formatter!)

Reply:
though should we consider using `std::chrono::system_clock` to get the time_t?